test/exdatatest.c

```diff
@@ -43,6 +50,127 @@ static void exfree(void *parent, void *ptr, CRYPTO_EX_DATA *ad,
     OPENSSL_assert(argp == saved_argp);
 }
 
+/*
+ * PRE-ALLOCATED EX_DATA IMPLEMENTATION
+ * Extended data structure is allocated in exnew2/freed in exfree2
+ * Data is stored inside extended data structure
+ */
+
+typedef struct myobj_ex_data_st {
+    char *hello;
+    int new;
+    int dup;
+} MYOBJ_EX_DATA;
+
+static void exnew2(void *parent, void *ptr, CRYPTO_EX_DATA *ad,
+          int idx, long argl, void *argp)
+{
+    int ret;
+    MYOBJ_EX_DATA *ex_data;
+
+    OPENSSL_assert(idx == saved_idx2);
+    if (idx != saved_idx2) {
```

Comment:
I thought I was following a paradigm used elsewhere, but now, I'm not so sure. I will correct.